atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -212,35 +247,36 @@ public void close() {
                     }
                 }
             }
+        } finally {
+            iter.close();
```

Comment:
yeah, I was thinking something along lines of change the arg to `rows` and use `try (ClosableIterator<AgnosticLightResultRow> iter = rows) {` with the main goal of having the improved close handling of try-with-resource so that if something throws in the main try block (say OOM or something else) that we don't drop that exception and any exceptions initiated from closing are added as [suppressed exceptions](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html#suppressed-exceptions) 


```
    private Map<Cell, Value> extractResults(
            DbReadTable table,
            ClosableIterator<AgnosticLightResultRow> rows) {
        Map<Cell, Value> results = Maps.newHashMap();
        Map<Cell, OverflowValue> overflowResults = Maps.newHashMap();
        try (ClosableIterator<AgnosticLightResultRow> iter = rows) {
            boolean hasOverflow = table.hasOverflowValues();
            while (iter.hasNext()) {
                AgnosticLightResultRow row = iter.next();
                Cell cell = Cell.create(row.getBytes("row_name"), row.getBytes("col_name"));
                Long overflowId = hasOverflow ? row.getLongObject("overflow") : null;
                if (overflowId == null) {
                    Value value = Value.create(row.getBytes("val"), row.getLong("ts"));
                    Value oldValue = results.put(cell, value);
                    if (oldValue != null && oldValue.getTimestamp() > value.getTimestamp()) {
                        results.put(cell, oldValue);
                    }
                } else {
                    OverflowValue ov = ImmutableOverflowValue.of(row.getLong("ts"), overflowId);
                    OverflowValue oldOv = overflowResults.put(cell, ov);
                    if (oldOv != null && oldOv.ts() > ov.ts()) {
                        overflowResults.put(cell, oldOv);
                    }
                }
            }
        }
        fillOverflowValues(table, overflowResults, results);
        return results;
    }
```